src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -751,6 +752,65 @@ class RandomDateGenerator : public Generator<bsoncxx::types::b_date> {
     const UniqueGenerator<int64_t> _maxGen;
 };
 
+class CycleGenerator : public Appendable {
+
+public:
+    CycleGenerator(const Node& node,
+                    GeneratorArgs generatorArgs,
+                    std::map<std::string, Parser<UniqueAppendable>> parsers)
+        : CycleGenerator(
+            extract(node, "of", "^Cycle"),
+            generatorArgs,
+            parsers,
+            extract(node, "number", "^Cycle").to<int64_t>()) {}
+
+    CycleGenerator(const Node& node,
+                GeneratorArgs generatorArgs,
```

Comment:
💅🏽 ensure you run `clang-format` prior to merge. We don't have CI for it (yet). I *think*/hope the instructions are in the readme.

Reply:
Updated this file. I don't think it's in the README though, is this the usual command we'll use for formatting?
`clang-format -style=file -i  src/value_generators/src/DocumentGenerator.cpp`

Reply:
I use this:

```sh
clang-format -i ./src/**/*.cpp ./src/**/*.hpp
```

I think the `**` requires zsh fwiw.